### ISSUE TITLE
Provide signatures for webhook alerts

### ISSUE THREAD

**Issue by nscuro** [DependencyTrack/dependency-track#1555]
### Current Behavior:

Alerts / notifications sent via Webhook are neither authenticated nor signed in any way. 
This makes it hard for receiving parties to verify whether a given notification was sent by DT.

### Proposed Behavior:

For the *Webhook* alert type, allow for an optional shared secret to be provided.
Before sending the webhook request, calculate an HMAC for the JSON payload, and include the resulting value in a request header (e.g. `X-Webhook-Signature`).

For reference, this is also how GitHub is doing it: https://docs.github.com/en/developers/webhooks-and-events/webhooks/securing-your-webhooks

**Comment by kkretsch** [comment]
As alternative an authentication header in the outgoing request might help too.
With this static auth I could trigger stuff to gitlab for example.

**Comment by mwerner96** [comment]
Hi, is there anything new on this topic? I would like to use the GitLab alert integration, but this webhook only works with an authorization key. I can't enter this key anywhere in the Dependency Track alerts.

**Comment by rroblik** [comment]
Any new on this feature? For security concern it could be cool and easier to implement than kafka sub client of hyades version  🙄

**Comment by jmezach** [comment]
Would be interested in this as well. I did notice we can send a header with a pre-configured value, so we can at least validate that on the receiving end. But having a signature would be nice of course.

**Comment by nscuro** [comment]
I agree that this is different from any form of `Authorization` header. This is primarily about integrity, not about passing credentials.
